common/sgx/tlsverifier.c

```diff
@@ -115,15 +116,33 @@ oe_result_t oe_verify_attestation_certificate(
     if (oe_cert_find_extension(
             &cert, (const char*)oid_oe_report, NULL, &report_size) !=
         OE_BUFFER_TOO_SMALL)
-        OE_RAISE(OE_FAILURE);
+    {
+        if (oe_cert_find_extension(
+                &cert, (const char*)oid_new_oe_report, NULL, &report_size) !=
+            OE_BUFFER_TOO_SMALL)
+        {
+            OE_RAISE(OE_FAILURE);
+        }
+        else
+        {
+            report = (uint8_t*)oe_malloc(report_size);
+            if (!report)
+                OE_RAISE(OE_OUT_OF_MEMORY);
+
+            OE_CHECK(oe_cert_find_extension(
+                &cert, (const char*)oid_new_oe_report, report, &report_size));
+        }
+    }
+    else
+    {
+        report = (uint8_t*)oe_malloc(report_size);
+        if (!report)
+            OE_RAISE(OE_OUT_OF_MEMORY);
 
-    report = (uint8_t*)oe_malloc(report_size);
-    if (!report)
-        OE_RAISE(OE_OUT_OF_MEMORY);
```

Comment:
I saw there are only two oids here so you didn't use a loop like you did for `oe_verify_attestation_certificate_with_evidence()`. And I suggest to use structure like:
```
if(check(A))
{ process(A); }
else if(check(B))
{ process(B); }
else
{ raise fail; }
```
That might make other readers easier to follow the code.